### ISSUE TITLE
Can't make predictions with model trained on a custom dataset

### ISSUE THREAD

**Issue by RibesM** [dbolya/yolact#132]
Hello,

I've trained a model with a yolact_base_config on a custom dataset. At the end of the training, the evaluation on the validation set looked like this:

```
Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 44.83 | 93.22 | 93.22 | 82.59 | 67.64 | 54.77 | 31.88 | 17.61 |  6.07 |  1.34 |  0.00 |
  mask | 48.29 | 82.46 | 80.34 | 78.50 | 76.32 | 66.83 | 44.57 | 30.41 | 19.19 |  4.31 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

[ 13]     780 || B: 0.214 | C: 0.103 | M: 0.925 | S: 0.031 | T: 1.273 || ETA: 0:00:26 || timer: 18.391
[ 13]     790 || B: 0.207 | C: 0.104 | M: 0.895 | S: 0.031 | T: 1.237 || ETA: 0:00:13 || timer: 1.078
```

However, when I try to display quantitative results on 1 image of the validation set with the saved model weights, with a command like this:
```
python eval.py --trained_model=weights/yolact_base_13_800.pth --score_threshold=0.3 --top_k=100 --image=data/datasets/val/20190719_070229_000760.jpg --display
```
I observe nothing on the output image. With some print functions, I found out that the variable `preds` in the function `evalimage` is `[None]`, and the variable `dets` in the function `postprocess` is therefore also `[None]`. This happens with 5 images from my validation set that I tested.
I don't understand why no prediction is made even though the evaluation went smoothly during training...

Any insight on what I could be doing wrong would be greatly appreciated.
And thank you for this great repository!

**Comment by dbolya** [comment]
Is your score threshold too high perhaps? Try removing `--score_threshold=0.3`. I will say that 800 iterations is definitely not enough for the network to learn properly. Perhaps you want to do some fine tuning instead? (#36)

As a sanity check, you might also want to try running eval on the validation set again, i.e.:
```python eval.py --trained_model=weights/yolact_base_13_800.pth```

p.s. You don't need to use `--display` when you're using `--image` so remove that too.

**Comment by RibesM** [comment]
Thank you for your suggestions.

I am trying fine tuning instead, and it looks much better indeed, but the evaluation's results seem to be fluctuating...
1st evaluation:

```

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 53.05 | 99.80 | 99.80 | 95.31 | 75.40 | 58.09 | 43.91 | 31.56 | 16.94 |  7.89 |  1.83 |
  mask | 78.58 | 96.35 | 95.38 | 95.38 | 95.38 | 93.36 | 90.41 | 84.97 | 78.94 | 46.48 |  9.13 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
```

And the 2nd evaluation:
```
       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 58.37 | 96.44 | 93.16 | 92.17 | 87.58 | 82.72 | 75.09 | 47.19 |  7.92 |  1.05 |  0.40 |
  mask | 73.82 | 91.96 | 90.91 | 90.91 | 87.07 | 87.07 | 81.05 | 75.64 | 74.02 | 55.72 |  3.85 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
```
Could it be overfitting? What learning rate would you suggest for a very small dataset (~70 images)?

**Comment by dbolya** [comment]
Well considering the box mAP increased, I'd say it's just variance (so might not be overfitting). 70 is a really small number of images so idk how well YOLACT would do.

As for learning rate, you should probably keep it at 1e-3 or it might actually overfit. Honestly, if the mAP is already this high might as well cut it off after you're satisfied with the qualitative performance. (btw did you get that working?)

**Comment by RibesM** [comment]
As you suggested, I cut it off at 200 steps with an evaluation like this:
 ```
       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 59.45 | 97.93 | 95.63 | 92.54 | 89.46 | 87.00 | 64.59 | 48.29 | 17.61 |  1.35 |  0.07 |
  mask | 79.55 | 95.63 | 95.63 | 95.63 | 91.39 | 91.39 | 87.86 | 86.77 | 83.39 | 61.99 |  5.80 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
```

Then I set the score threshold at 0.1 and tried the qualitative evaluation again. Now it does display some masks and boxes! But at positions completely unrelated with the initial targets (the predicted masks appeared on the background).

At this point it might be a problem with the changes I made to account for the discrepancies between my annotations and the COCO format, so I will close this issue. Thank you very much for your help!
